ruby/tests/basic.rb

```diff
@@ -62,572 +61,600 @@ module BasicTest
+    l.clear
+    assert_equal l.count, 0
+    l += [1, 2, 3, 4]
+    l.replace([5, 6, 7, 8])
+    assert_equal l, [5, 6, 7, 8]
 
-      l4 = Google::Protobuf::RepeatedField.new(:int32)
-      l4[5] = 42
-      assert l4 == [0, 0, 0, 0, 0, 42]
+    l4 = Google::Protobuf::RepeatedField.new(:int32)
+    l4[5] = 42
+    assert_equal [0, 0, 0, 0, 0, 42], l4
 
-      l4 << 100
-      assert l4 == [0, 0, 0, 0, 0, 42, 100]
-      l4 << 101 << 102
-      assert l4 == [0, 0, 0, 0, 0, 42, 100, 101, 102]
+    l4 << 100
+    assert_equal l4, [0, 0, 0, 0, 0, 42, 100]
+    l4 << 101 << 102
+    assert_equal l4, [0, 0, 0, 0, 0, 42, 100, 101, 102]
+  end
+
+  def test_rptfield_msg
+    l = Google::Protobuf::RepeatedField.new(:message, TestMessage)
+    l.push TestMessage.new
+    assert_equal 1, l.count
+    assert_raises TypeError do
+      l.push TestMessage2.new
+    end
+    assert_raises TypeError do
+      l.push 42
     end
 
-    def test_rptfield_msg
-      l = Google::Protobuf::RepeatedField.new(:message, TestMessage)
-      l.push TestMessage.new
-      assert l.count == 1
-      assert_raise TypeError do
-        l.push TestMessage2.new
-      end
-      assert_raise TypeError do
-        l.push 42
-      end
+    l2 = l.dup
+    assert_equal l2[0], l[0]
+    assert_equal l2[0].object_id, l[0].object_id
+
+    l2 = Google::Protobuf.deep_copy(l)
+    assert_equal l2[0], l[0]
+    refute_equal l2[0].object_id, l[0].object_id
+
+    l3 = l + l2
+    assert_equal 2, l3.count
+    assert_equal l3[0], l[0]
+    assert_equal l3[1], l2[0]
+    l3[0].optional_int32 = 1000
+    assert_equal 1000, l[0].optional_int32
+
+    new_msg = TestMessage.new(:optional_int32 => 200)
+    l4 = l + [new_msg]
+    assert_equal 2, l4.count
+    new_msg.optional_int32 = 1000
+    assert_equal 1000, l4[1].optional_int32
+  end
 
-      l2 = l.dup
-      assert l2[0] == l[0]
-      assert l2[0].object_id == l[0].object_id
-
-      l2 = Google::Protobuf.deep_copy(l)
-      assert l2[0] == l[0]
-      assert l2[0].object_id != l[0].object_id
-
-      l3 = l + l2
-      assert l3.count == 2
-      assert l3[0] == l[0]
-      assert l3[1] == l2[0]
-      l3[0].optional_int32 = 1000
-      assert l[0].optional_int32 == 1000
-
-      new_msg = TestMessage.new(:optional_int32 => 200)
-      l4 = l + [new_msg]
-      assert l4.count == 2
-      new_msg.optional_int32 = 1000
-      assert l4[1].optional_int32 == 1000
-    end
-
-    def test_rptfield_enum
-      l = Google::Protobuf::RepeatedField.new(:enum, TestEnum)
-      l.push :A
-      l.push :B
-      l.push :C
-      assert l.count == 3
-      assert_raise NameError do
-        l.push :D
-      end
-      assert l[0] == :A
+  def test_rptfield_enum
+    l = Google::Protobuf::RepeatedField.new(:enum, TestEnum)
+    l.push :A
+    l.push :B
+    l.push :C
+    assert_equal 3, l.count
+    assert_raises NameError do
+      l.push :D
+    end
+    assert_equal :A, l[0]
 
+    # enum value has to be defined in java
```

Comment:
Hmm, has proto3 support for Java not yet been implemented/released? For proto3, the semantics of enums are supposed to be open, but of course if protobuf-Java hasn't been augmented to support this yet then this won't work yet.